### PR TITLE
Changes on function/metric loggers

### PR DESCRIPTION
**Pull request by pedronasser**
Make them more plugglable

### CODE REVIEW

File: api/runner/async_runner.go [hunk 1]
```diff
@@ -52,8 +51,7 @@ func getCfg(task *models.Task) *Config {
 		Timeout: time.Duration(*task.Timeout) * time.Second,
 		ID:      task.ID,
 		AppName: task.AppName,
-		Stdout:  stderr,
-		Stderr:  stderr,
+		Stdout:  ioutil.Discard,
```
Comment (ccirello) [inline api/runner/async_runner.go]: Is the intention really to discard Stdout? I understand that stderr is being plugged later down the process, but stdout not.

`getCfg()` is used by `startAsyncRunners()` in order to connect both stdout and stderr. Could you please check that this behavior hasn't been changed?

Comment (pedronasser) [inline api/runner/async_runner.go]: @ccirello If read the previous impl of `getCfg()` you'll see that we was sending the output to the stderr used to log. What is the benefit of logging the async result?

Comment (ccirello) [inline api/runner/async_runner.go]: > What is the benefit of logging the async result?

So you can actually inspect what an async is doing. Or putting differently, this is a subtle change that I wouldn't mind as long as we make clear that this is the intention.

Comment (pedronasser) [inline api/runner/async_runner.go]: Oh my bad.
I'll describe it better.

Comment (ccirello) [inline api/runner/async_runner.go]: @treeder is this change OK? I know you have a more informed view on what to do with async stdout output.

**Pull request merged**